docs/source/llms/tracing/index.rst

```diff
@@ -0,0 +1,558 @@
+
+.. code-block:: python
+
+    import mlflow
+
+
+    @mlflow.trace
+    def first_func(x, y=2):
+        return x + y
+
+
+    @mlflow.trace
+    def second_func(a, b=3):
+        return a * b
+
+
+    def do_math(a, x, operation="add"):
+        # Use the fluent API context handler to create a new span
+        with mlflow.start_span(name="Math") as span:
+            # Specify the inputs and attributes that will be associated with the span
+            span.set_inputs({"a": a, "x": x})
+            span.set_attributes({"mode": operation})
+
+            # Both of these functions are decorated for tracing and will be associated
+            # as 'children' of the parent 'span' defined with the context handler
+            first = first_func(x)
+            second = second_func(a)
+
+            result = None
+
+            if operation == "add":
+                result = first + second
+            elif operation == "subtract":
+                result = first - second
+            else:
+                raise AttributeError(f"Unsupported Operation Mode: {operation}")
+
+            # Specify the output result to the span
+            span.set_outputs({"result": result})
+
+            return result
+
+When calling the ``do_math`` function, a trace will be generated that has the root span (parent) defined as the 
+context handler ``with mlflow.start_span():`` call. The ``first_func`` and ``second_func`` calls will be associated as child spans
+to this parent span due to the fact that they are both decorated functions (having ``@mlflow.trace`` decorated on the function definition). 
+
+Running the following code will generate a trace. 
+
+.. code-block:: python
+
+    do_math(8, 3, "add")
+
+This trace can be seen within the MLflow UI:
+
+.. figure:: ../../_static/images/llms/tracing/trace-view.png
+    :alt: Trace within the MLflow UI 
+    :width: 100%
+    :align: center
+
+If an `Exception` is raised during processing of a trace-instrumented operation, an indication will be shown within the UI that the invocation was not 
+successful and a partial capture of data will be available to aid in debugging, as shown below:
+
+.. code-block:: python
+
+    # This will raise an AttributeError exception
+    do_math(3, 2, "multiply")
+
+.. figure:: ../../_static/images/llms/tracing/trace-error.png
+    :alt: Trace Error
+    :width: 100%
+    :align: center
+
+Function wrapping
+#################
+
+Function wrapping provides a flexible way to add tracing to existing functions without modifying their definitions. This is particularly useful when 
+you want to add tracing to third-party functions or functions defined outside of your control. By wrapping an external function with :py:func:`mlflow.trace`, you can
+capture its inputs, outputs, and execution context.
+
+
+.. code-block:: python
+
+    import mlflow
+
+
+    def my_external_function(x, y):
+        return x + y
+
+
+    def invocation(x, y=4):
```

Comment:
nit: (1) Do we need the parent span in this example? (2) Can we import some built-in function instead of defining custom function?

Reply:
We could, but I almost feel like wrapping a builtin language function would be a bit distracting.

Reply:
True, let's keep the original example, then!